scapy/arch/windows/__init__.py

```diff
@@ -824,22 +876,30 @@ def _read_routes6_post2008():
         _append_route6(routes6, dpref, dp, nh, iface, lifaddr, metric)
     return routes6
 
-def _get_i6_metric(index):
+def _get_i6_metric():
     # Returns the INTERFACE metric from the interface index
-    ps = sp.Popen([conf.prog.cmd, '/c', 'netsh interface ipv6 show interfaces level=verbose ' + index], stdout = sp.PIPE)
-    stdout, stdin = ps.communicate()
-    stdout = stdout.decode("utf8", "ignore")
-    metric_line = stdout.split('\n')[6]
-    metric = re.search(re.compile(".*:\s+(\d+)"), metric_line).group(1)
-    return int(metric)
+    query_cmd = "netsh interface ipv6 show interfaces level=verbose"
+    stdout = POWERSHELL_PROCESS.query([query_cmd])
+    res = {}
+    _buffer = []
+    for _line in stdout:
+        if not _line.strip():
+            continue
+        _buffer.append(_line)
+        if len(_buffer) == 32: # An interface, with all parameters, is 32 lines long
+            if_index = re.search(re.compile(".*:\s+(\d+)"), _buffer[3]).group(1)
+            if_metric = int(re.search(re.compile(".*:\s+(\d+)"), _buffer[5]).group(1))
```

Comment:
re.compile is called twice with the same arguments. And it's called in the loop.

Please move outside of the loop, assign meaningful name (<xxx>_pattern  = re.compile("......")), then just use inside the loop.